src/Microsoft.AspNetCore.Sockets.Abstractions/ConnectionMetadata.cs

```diff
@@ -3,14 +3,15 @@
 
 using System;
 using System.Collections;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 
 namespace Microsoft.AspNetCore.Sockets
 {
     internal class ConnectionMetadata : IDictionary<object, object>
     {
         public ConnectionMetadata()
-            : this(new Dictionary<object, object>())
+            : this(new ConcurrentDictionary<object, object>())
```

Comment:
Change is because of https://github.com/aspnet/SignalR/issues/1636

Functional tests were sporadically failing because of the error there